searx/engines/livespace.py

```diff
@@ -0,0 +1,118 @@
+# SPDX-License-Identifier: AGPL-3.0-or-later
+"""
+ LiveSpace (Videos)
+"""
+
+from urllib.parse import quote_plus
+from json import loads
+from datetime import datetime
+
+# about
+about = {
+    "website": 'https://live.space',
+    "wikidata_id": None,
+    "official_api_documentation": None,
+    "use_official_api": True,
+    "require_api_key": False,
+    "results": 'JSON',
+}
+
+categories = ['videos']
+
+engine_url = 'https://backend.live.space/search/public/stream' + '?page={page}&searchKey={query}&size={size}'
+
+# engine dependent config
+paging = True
+number_of_results = 10
+
+
+def request(query, params):
+    # basic search
+    page = params['pageno'] - 1
+
+    string_args = dict(query=quote_plus(query), page=page, size=number_of_results)
+
+    params['url'] = engine_url.format(**string_args)
+
+    params['headers'] = {'Accept': 'application/json', 'Content-Type': 'application/json'}
+
+    return params
+
+
+def response(resp):
+    results = []
+
+    print(resp.text)
+
+    search_results = loads(resp.text)
+
+    items = search_results.get('result', [])
+
+    for result in items:
+        # for live videos
+        now = datetime.now()
+
+        # stream start
+        startTimeStamp_string = result.get("startTimeStamp", None)
+        startTimeStamp = (
+            None if startTimeStamp_string is None else datetime.utcfromtimestamp(float(startTimeStamp_string) / 1000)
+        )
+
+        # stream title
+        title = result.get("title", None)
+
+        user_obj = result.get("user", None)
+
+        if user_obj is not None:
+            # username
+            username = user_obj.get("userName", None)
+
+        # stream thumbnail
+        thumbnailUrl = result.get("thumbnailUrl", None)
+
+        # stream category
+        category = result.get("category/name", None)
+
+        # stream tags
+        bad_prefix = 'null;'
+        tags = result.get("tags", None)
+        if tags is not None and tags[: len(bad_prefix)] == bad_prefix:
+            tags = tags[len(bad_prefix) :]
+
+        # for VODs (videos on demand)
+        endTimeStamp_string = result.get("endTimeStamp", None)
+        end = None if endTimeStamp_string is None else datetime.utcfromtimestamp(float(endTimeStamp_string) / 1000)
+
+        content = []
+        if category is not None:
+            content.append(f'category - {category}')
+        if tags is not None:
+            content.append(f'tags - {tags}')
+
+        url = None if username is None else f'https://live.space/watch/{username}'
+
+        delta = (now if end is None else end) - startTimeStamp
+        total_seconds = int(delta.total_seconds())
+        hours, remainder = divmod(total_seconds, 60 * 60)
+        minutes, seconds = divmod(remainder, 60)
+        if hours > 0:
+            timestring = f"{hours}h {minutes:02}m {seconds:02}s"
+        elif minutes > 0:
+            timestring = f"{minutes:02}m {seconds:02}s"
+        else:
+            timestring = f"{seconds:02}s"
```

Comment:
I'm not sure if we already have a pre-build function for formatting time span strings. (maybe @return42 knows? :) )

If not, we should probably consider adding it soon, as I've already seen multiple engines formatting similar time strings.

Reply:
would you like me to raise this as an issue and create a new PR?

Reply:
Yes, that's a good idea :+1: